pkg/build/admission/defaults/admission.go

```diff
@@ -71,6 +71,28 @@ func (b BuildDefaults) applyPodDefaults(pod *kapi.Pod) {
 	for k, v := range b.config.Annotations {
 		addDefaultAnnotations(k, v, pod.Annotations)
 	}
+
+	// Apply default resources
+	defaultResources := b.config.Resources
+	if len(pod.Spec.Containers[0].Resources.Limits) == 0 {
```

Comment:
> the length of a nil map is zero

True; I guess my point was that the length of a non-nil but empty map (i.e. not needing initialisation) is also zero ;-)